### PR TITLE
Use new format for logging exceptions - plugins/dns.py

### PR DESCRIPTION
**Pull request by vighnesh-kadam**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:#3156
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/plugins/dns.py [hunk 1]
```diff
@@ -211,17 +210,15 @@ async def start(self) -> None:
...
 
     async def stop(self) -> None:
         """Stop CoreDNS."""
         _LOGGER.info("Stopping CoreDNS plugin")
         try:
             await self.instance.stop()
         except DockerError as err:
-            _LOGGER.error("Can't stop CoreDNS plugin")
-            raise CoreDNSError() from err
+            raise CoreDNSError(f"Can't stop CoreDNS plugin", _LOGGER.error) from err
```
Comment (ludeeus) [inline supervisor/plugins/dns.py:221]:
```suggestion
            raise CoreDNSError("Can't stop CoreDNS plugin", _LOGGER.error) from err
```

File: supervisor/plugins/dns.py [hunk 2]
```diff
@@ -211,17 +210,15 @@ async def start(self) -> None:
         try:
             await self.instance.run()
         except DockerError as err:
-            _LOGGER.error("Can't start CoreDNS plugin")
-            raise CoreDNSError() from err
+            raise CoreDNSError(f"Can't start CoreDNS plugin", _LOGGER.error) from err
```
Comment (ludeeus) [inline supervisor/plugins/dns.py:213]:
```suggestion
            raise CoreDNSError("Can't start CoreDNS plugin", _LOGGER.error) from err
```

File: supervisor/plugins/dns.py [hunk 3]
```diff
@@ -199,8 +199,7 @@ async def restart(self) -> None:
         try:
             await self.instance.restart()
         except DockerError as err:
-            _LOGGER.error("Can't start CoreDNS plugin")
-            raise CoreDNSError() from err
+            raise CoreDNSError(f"Can't start CoreDNS plugin", _LOGGER.error) from err
```
Comment (ludeeus) [inline supervisor/plugins/dns.py:202]:
```suggestion
            raise CoreDNSError("Can't start CoreDNS plugin", _LOGGER.error) from err
```

File: supervisor/plugins/dns.py [hunk 4]
```diff
@@ -178,8 +178,8 @@ async def update(self, version: Optional[AwesomeVersion] = None) -> None:
         try:
             await self.instance.update(version, image=self.sys_updater.image_dns)
         except DockerError as err:
-            _LOGGER.error("CoreDNS update failed")
-            raise CoreDNSUpdateError() from err
+            raise CoreDNSUpdateError(f"CoreDNS update failed", _LOGGER.error) from err
```
Comment (ludeeus) [inline supervisor/plugins/dns.py:181]:
```suggestion
            raise CoreDNSUpdateError("CoreDNS update failed", _LOGGER.error) from err
```

**Review by ludeeus — CHANGES_REQUESTED** [review]

File: supervisor/plugins/dns.py (hunk 4 above)
Comment (vighnesh-kadam) [reply supervisor/plugins/dns.py:181]: Sorry for that mistake !! Will take care henceforth, thanks!

File: supervisor/plugins/dns.py [hunk 5]
```diff
@@ -178,8 +178,8 @@ async def update(self, version: Optional[AwesomeVersion] = None) -> None:
         try:
             await self.instance.update(version, image=self.sys_updater.image_dns)
         except DockerError as err:
-            _LOGGER.error("CoreDNS update failed")
-            raise CoreDNSUpdateError() from err
+            raise CoreDNSUpdateError("CoreDNS update failed", _LOGGER.error) from err
+
```
Comment (ludeeus) [inline supervisor/plugins/dns.py:182]:
```suggestion
```

**Review by ludeeus — APPROVED** [review]
Thanks :+1:

**Pull request merged**